### PR TITLE
Additional parameter options for building Tulsi

### PR DESCRIPTION
**Pull request by maxwellE**
Allows users to specify where Tulsi should be unzipped to after install using the `-d` option and allows them to specify the Bazel path to use with the `-b` option

### CODE REVIEW

**Comment by maxwellE** [comment]
@googlebot I signed it!

File: build_and_run.sh [hunk 1]
```diff
@@ -21,11 +21,22 @@
 
 set -eu
 
-readonly unzip_dir="${1:-$HOME/Applications}"
+unzip_dir="$HOME/Applications"
+bazel_path="bazel"
+
+while [ $# -gt 0 ]; do
+
+   if [[ $1 == *"--"* ]]; then
```
Comment (DavidGoldman) [inline build_and_run.sh:29]: Missed this before, this seems a bit odd (shouldn't need the beginning *, right)?

WDYT about instead using `getopts` with short form args (I guess `-b` for bazel and `-d` for dir) and then you can also have a help option?

Comment (tinder-maxwellelliott) [reply build_and_run.sh:29]: Sounds good!

File: README.md [hunk 2]
```diff
@@ -13,7 +13,12 @@ plant.
     locally. If it isn't, feel free to remove the `--xcode_version` flag or
     modify it as you wish, but note that Tulsi may not build correctly with
     different versions of Xcode.
-1.  Run `build_and_run.sh`. This will install Tulsi.app inside ~/Applications.
+2.  Run `build_and_run.sh`. This will install Tulsi.app inside ~/Applications.
```
Comment (DavidGoldman) [inline README.md:16]: by default?

File: README.md [hunk 3]
```diff
@@ -13,7 +13,12 @@ plant.
...
+
+**Supported Options:**
+
+* `-b`: Bazel path that Tulsi should use to build and install the app (Default is `bazel`)
```
Comment (DavidGoldman) [inline README.md:20]: binary path maybe?

File: README.md [hunk 4]
```diff
@@ -13,7 +13,12 @@ plant.
...
+* `-d`: Location will unzip the built Tulsi app (Default is `$HOME/Applications`)
```
Comment (DavidGoldman) [inline README.md:21]: Where to? maybe the folder where to install the Tulsi app into?

File: build_and_run.sh [hunk 5]
```diff
@@ -21,11 +21,31 @@
...
+unzip_dir="$HOME/Applications"
+bazel_path="bazel"
+
+while getopts ":b:d:h" opt; do
+  case ${opt} in
+    h)
+      echo "Usage:"
+      echo "    ./build_and_run -h          Display this help message."
+      echo "    ./build_and_run -b PATH     Bazel path used to build Tulsi"
+      echo "    ./build_and_run -d PATH     Unzip built Tulsi App to a provided Path"
```
Comment (DavidGoldman) [inline build_and_run.sh:33]: install is probably more clear

**Comment by maxwellE** [comment]
> Looks good, will merge this later today

Thank you. I can look into also adding the ability to override the xcode version using the flags provided, associated issue #99

File: README.md [hunk 6]
```diff
@@ -13,7 +13,13 @@ plant.
...
     modify it as you wish, but note that Tulsi may not build correctly with
     different versions of Xcode.
-1.  Run `build_and_run.sh`. This will install Tulsi.app inside ~/Applications.
+2.  Run `build_and_run.sh`. This will install Tulsi.app inside `$HOME/Applications` by default.
+
+**Supported Options:**
+
+* `-b`: Bazel binary that Tulsi should use to build and install the app (Default is `bazel`)
+* `-d`: The folder where to install the Tulsi app into (Default is `$HOME/Applications`)
+* `-x`: The XCode version Tulsi should be built for (Default is `11.3.1`)
```
Comment (DavidGoldman) [inline README.md:22]: nit: Xcode

File: build_and_run.sh [hunk 7]
```diff
@@ -21,11 +21,34 @@
...
+xcode_version="11.3.1"
+
+while getopts ":b:d:x:h" opt; do
+  case ${opt} in
+    h)
+      echo "Usage:"
+      echo "    ./build_and_run -h          Display this help message."
+      echo "    ./build_and_run -b PATH     Bazel binary used to build Tulsi"
+      echo "    ./build_and_run -d PATH     Intall Tulsi App at the provided path"
+      echo "    ./build_and_run -x VERSION  XCode version Tulsi should be built for"
```
Comment (DavidGoldman) [inline build_and_run.sh:35]: Xcode again

**Pull request closed without merging**